src/include/migraphx/op/allocate.hpp

```diff
@@ -48,9 +48,9 @@ namespace op {
  */
 struct allocate
 {
-    shape s{};
+    std::optional<shape> s;
     // for dynamic allocate to set the buffer type
-    shape::type_t buf_type = shape::half_type;
+    std::optional<shape::type_t> buf_type;
```

Comment:
These seems mutually exclusive. Maybe we should use `std::variant` instead , but we will need to update the serialization to support variant.